builder/triton/ssh.go

```diff
@@ -0,0 +1,36 @@
+package triton
+
+import (
+	"fmt"
+
+	"github.com/mitchellh/multistep"
+	"golang.org/x/crypto/ssh"
+)
+
+func commHost(state multistep.StateBag) (string, error) {
+	driver := state.Get("driver").(Driver)
+	machineID := state.Get("machine").(string)
+
+	machine, err := driver.GetMachine(machineID)
+	if err != nil {
+		return "", err
+	}
+
+	return machine, nil
+}
+
+func sshConfig(state multistep.StateBag) (*ssh.ClientConfig, error) {
+	config := state.Get("config").(Config)
+
+	signer, err := ssh.ParsePrivateKey([]byte(config.Comm.SSHPrivateKey))
+	if err != nil {
+		return nil, fmt.Errorf("Error setting up SSH config: %s", err)
+	}
+
+	return &ssh.ClientConfig{
+		User: config.Comm.SSHUsername,
+		Auth: []ssh.AuthMethod{
+			ssh.PublicKeys(signer),
+		},
+	}, nil
```

Comment:
I did some research and I don't know if adding functionality like SSH agent support and password are really that big of a deal for the Triton builder to support in phase one. Here is why:

You must always add an SSH RSA key to your Triton account . This key is used when accessing the Triton Cloud API (REST API to access Triton). You cannot access the Cloud API without it (it is used to additionally sign requests you make over HTTPS). Every smartmachine you spin up automatically has the root user configured in such a way you can log in with this key. The key is not just added to the `authorized_key`'s, there is a special Triton SSH plugin in OpenSSH which ensures the keys added via the Cloud API / Operations portal can be used in that VM. 

In order to use for example password authentication it would require to first create a new base image which would support authentication methods like password or have different SSH keys in it besides the one set by Triton. As far as I know none of the Joyent provided images do that and that seems like a more advanced use case to me.

Also the Triton Go Cloud API Library provided by Joyent can currently only deal with private keys in files. So unless we add ssh agent support to that library it will always require the private key to also sit in an file (regardless if we use an SSH agent to connect to the VM or not).